WordPress/Classes/ViewRelated/Blog/SiteManagement/SiteTagsViewController.swift

```diff
@@ -0,0 +1,376 @@
+            self?.refreshControl?.endRefreshing()
+            self?.refreshNoResultsView()
+        }) { [weak self] error in
+            self?.tagsFailedLoading(error: error)
+        }
+    }
+
+    private func setupNavigationBar() {
+        configureRightButton()
+    }
+
+    private func configureRightButton() {
+        navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .add,
+                                                            target: self,
+                                                            action: #selector(createTag))
+    }
+
+    private func setupSearchBar() {
+       tableView.tableHeaderView = searchController.searchBar
+    }
+
+    private func removeSearchBar() {
+        tableView.tableHeaderView = nil
+    }
+
+    @objc private func createTag() {
+        navigate(to: nil)
+    }
+
+    private func refreshNoResultsView() {
+        guard resultsController.fetchedObjects?.count == 0 else {
+            noResultsView.removeFromSuperview()
+            setupSearchBar()
+            tableView.reloadData()
+            return
+        }
+
+        if isPerformingInitialSync {
+            setupLoadingView()
+        } else {
+            setupEmptyResultsView()
+        }
+
+        if noResultsView.superview == nil {
+            tableView.addSubview(withFadeAnimation: noResultsView)
+        }
+
+        removeSearchBar()
+    }
+
+    private func setupLoadingView() {
+        noResultsView.accessoryView = nil
+        noResultsView.titleText = loadingMessage()
+        noResultsView.messageText = ""
+        noResultsView.buttonTitle = ""
+    }
+
+    private func setupEmptyResultsView() {
+        noResultsView.accessoryView = noResultsAccessoryView()
+        noResultsView.titleText = noResultsTitle()
+        noResultsView.messageText = noResultsMessage()
+        noResultsView.buttonTitle = noResultsButtonTitle()
+        noResultsView.button.addTarget(self, action: #selector(createTag), for: .touchUpInside)
+    }
+
+    func tagsFailedLoading(error: Error) {
+        DDLogError("Tag management. Error loading tags for \(String(describing: blog.url)): \(error)")
+    }
+}
+
+// MARK: - Table view datasource
+extension SiteTagsViewController {
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return resultsController.sections?.count ?? 0
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return resultsController.sections?[section].numberOfObjects ?? 0
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: TableConstants.cellIdentifier, for: indexPath) as? TitleBadgeDisclosureCell, let tag = tagAtIndexPath(indexPath) else {
+            return TitleBadgeDisclosureCell()
+        }
+
+        cell.name = tag.name
```

Comment:
I think this should have html entities decoded? I did a test with a tag name containing an `&` and it became `&amp;` and displayed that way in the UI, but not on the web.

fwiw, this seems to also be an existing problem with the tags list in the blog posts list.